detector/interface.go

```diff
@@ -27,6 +27,18 @@ type MasterChanged interface {
 	OnMasterChanged(*mesos.MasterInfo)
 }
 
+// optional interface that, if implemented by the same object as implements
```

Comment:
If we follow Godoc conventions, documentation is gonna be nice. Let's start this comment with the name of the entity it describes. `AllMasters defines ...`